packages/ui/src/providers/keyring/provider.tsx

```diff
@@ -15,21 +15,46 @@ function isKeyringLoaded() {
   }
 }
 
+const loadKeysFromExtension = async () => {
+  await web3Enable('Pioneer')
+  const injectedAccounts = await web3Accounts()
+  const allAccounts = injectedAccounts.map(({ address, meta }) => ({
+    address,
+    meta: { ...meta, name: `${meta.name} (${meta.source})` },
+  }))
+
+  if (!isKeyringLoaded()) {
+    keyring.loadAll({ isDevelopment: true }, allAccounts)
+  }
+}
+
+// Extensions is not always ready on application load, hence the check
+const onExtensionLoaded = (callback: () => void) => () => {
+  const intervalId = setInterval(() => {
+    if (Object.keys((window as any).injectedWeb3).length) {
+      clearInterval(intervalId)
+      callback()
+    }
+  }, 20)
+
+  return () => clearInterval(intervalId)
+}
+
 export const KeyringContextProvider = (props: Props) => {
+  const [isLoaded, setIsLoaded] = useState(false)
+
+  useEffect(
+    onExtensionLoaded(() => setIsLoaded(true)),
+    []
+  )
+
   useEffect(() => {
-    web3Enable('Pioneer')
-      .then(() => web3Accounts())
-      .then((injectedAccounts) => {
-        const allAccounts = injectedAccounts.map(({ address, meta }) => ({
-          address,
-          meta: { ...meta, name: `${meta.name} (${meta.source})` },
-        }))
-
-        if (!isKeyringLoaded()) {
-          keyring.loadAll({ isDevelopment: true }, allAccounts)
-        }
-      })
-  }, [])
+    if (!isLoaded) {
+      return
+    }
+
+    loadKeysFromExtension()
```

Comment:
It's async, so perhaps:
```suggestion
    loadKeysFromExtension().catch(console.error)
```